homeassistant/components/buienradar/config_flow.py

```diff
@@ -0,0 +1,167 @@
+"""Config flow for buienradar integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DOMAIN, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_CAMERA,
+    CONF_COUNTRY,
+    CONF_DIMENSION,
+    CONF_SENSOR,
+    CONF_WEATHER,
+    DEFAULT_DIMENSION,
+    DOMAIN,
+    HOME_LOCATION_NAME,
+    SUPPORTED_COUNTRY_CODES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured buienradar instances."""
+    entries = []
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data.get(CONF_CAMERA):
+            entries.append(
+                f"{entry.data.get(CONF_DIMENSION)}-{entry.data.get(CONF_COUNTRY)}"
+            )
+        else:
+            entries.append(
+                f"{entry.data.get(CONF_LATITUDE)}-{entry.data.get(CONF_LONGITUDE)}"
+            )
+    return set(entries)
+
+
+class BuienradarFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for buienradar."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            if user_input[CONF_DOMAIN] == "Camera":
+                return await self.async_step_camera()
+
+            return await self.async_step_weather()
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_DOMAIN): vol.In(["Camera", "Weather"]),
```

Comment:
The main reason to separate is that the camera and weather platform poll a completely different API.

In hindsight, I think disabling an entity would be a good alternative for users who e.g. don't want their device to poll a camera endpoint.

Reply:
Maybe one thing, the weather API is useless for Belgian users if I remember correctly.

Reply:
I'm open to address changes requested to a reasonable level of effort, but if the answer is split it into multiple integrations I have to abandon this and leave it in it's current state...

Reply:
> but if the answer is split it into multiple integrations I have to abandon this and leave it in it's current state...

That was the suggestion yes. To be honest, that would make the code a lot easier as well.

Reply:
I can follow the reasoning and do not disagree on that, I'm just saying I'm not going to do that. I've updated the existing integration, where everything is in one integration, to get rid of platform setup and yaml. Imho, this PR is not the smallest one to avoid breaking changes, but aside from that the config flow code is overseeable and I think beneficial for new users stepping into Home Assistant who are not familiar with yaml. If that's not accepted in it's current form, than it is what it is and we'll leave it as is until someone may have the time to completely revise this.

Reply:
Having this option in the config flow is weird and should be removed. I would instead always add the camera entity but have it be disabled by default. The only sensor that should be enabled by default is the weather one.

Reply:
Just got some input from @MartinHjelmare on Discord as well. Remove the camera from the config flow, create a camera entity, and disable it by default. Put the camera country selector in the options flow so one can change it.

Reply:
Thanks for the input, I can work with that.